<launch>
  <group ns="rtabmap">

    <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap" output="screen">
      
      <!-- MODE ABONNEMENT -->
      <param name="subscribe_rgbd" type="bool" value="false"/>
      <param name="subscribe_rgb" type="bool" value="true"/>
      <param name="subscribe_scan" type="bool" value="true"/>
      <param name="subscribe_stereo" type="bool" value="false"/>
      <param name="approx_sync" type="bool" value="true"/>
      
      <!-- FRAME DE RÉFÉRENCE -->
      <param name="frame_id" type="string" value="robot_base_link"/>
      <param name="queue_size" type="int" value="30"/>
      <param name="sync_queue_size" type="int" value="50"/>

      <!-- REMAP POUR RGB + DEPTH -->
      <remap from="rgb/image"       to="/robot/front_rgbd_camera/color/image_raw"/>
      <remap from="depth/image"     to="/robot/front_rgbd_camera/depth/image_raw"/>
      <remap from="rgb/camera_info" to="/robot/front_rgbd_camera/color/camera_info"/>
      
      <!-- AUTRES REMAPS -->
      <remap from="scan" to="/robot/merged_laser/scan"/>
      <remap from="odom" to="/robot/robotnik_base_control/odom"/>

      <!-- PARAMÈTRES SLAM -->
      <param name="RGBD/NeighborLinkRefining" type="string" value="true"/>
      <param name="RGBD/ProximityBySpace"     type="string" value="true"/>
      <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
      <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
      <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>
      <param name="Grid/FromDepth"            type="string" value="true"/>
      <param name="Reg/Force3DoF"             type="string" value="true"/>
      <param name="Reg/Strategy"              type="string" value="1"/>
      <param name="Icp/VoxelSize"             type="string" value="0.05"/>

    </node>

  </group>
</launch>
